shared-libs/transitions/src/transitions/create_user_for_contacts.js

```diff
@@ -0,0 +1,127 @@
+
+  return `${username}-${randomNum}`;
+};
+
+const generateUniqueUsername = (contactName) => {
+  const username = generateUsername(contactName);
+  return db.users.get(`org.couchdb.user:${username}`)
+    .then(() => generateUniqueUsername(contactName))
+    .catch(error => {
+      if (error.status === 404) {
+        // this username is available
+        return username;
+      }
+      throw error;
+    });
+};
+
+const createNewUser = ({ roles }, { _id, name, phone, parent }) => {
+  if (!name) {
+    throw new Error(`Replacement contact [${_id}] must have a name.`);
+  }
+  return generateUniqueUsername(name)
+    .then(username => {
+      const user = {
+        username,
+        token_login: true,
+        roles,
+        phone,
+        place: parent._id,
+        contact: _id,
+        fullname: name,
+      };
+      return users.createUser(user, config.get('app_url'))
+        .catch(err => {
+          if (!err.message || typeof err.message === 'string') {
+            throw err;
+          }
+          const message = err.message.message ? err.message.message : err.message;
+          throw new Error(`Error creating new user: ${JSON.stringify(message)}`);
+        });
+    });
+};
+
+const replaceUser = (originalContact) => {
+  const { _id, user_for_contact : { replaced: { by } } } = originalContact;
+  return Promise
+    .all([
+      getNewContact(by),
+      users.getUserSettings({ contact_id: _id }),
+    ])
+    .then(([newContact, originalUserSettings]) => {
+      return createNewUser(originalUserSettings, newContact)
+        .then(() => users.deleteUser(originalUserSettings.name));
+    })
+    .then(() => originalContact.user_for_contact.replaced.status = UserCreationStatus.COMPLETE);
+};
+
+/**
+ * Replace a user whose contact has been marked as ready to be replaced with a new user.
+ */
+module.exports = {
+  name: NAME,
+  init: () => {
+    const tokenLogin = config.get('token_login');
+    if (!tokenLogin || !tokenLogin.enabled) {
+      throw new Error(
+        `Configuration error. Token login must be enabled to use the create_user_for_contacts transition.`
+      );
+    }
+
+    const appUrl = config.get('app_url');
+    if(!appUrl) {
+      throw new Error(
+        'Configuration error. The app_url must be defined to use the create_user_for_contacts transition.'
+      );
+    }
+  },
+  filter: (doc) => {
+    const contactType = contactTypeUtils.getContactType(config.getAll(), doc);
+    if (!contactType || !contactTypeUtils.isPersonType(contactType)) {
+      return false;
+    }
+
+    return doc.user_for_contact &&
+      doc.user_for_contact.replaced &&
+      doc.user_for_contact.replaced.status === UserCreationStatus.READY;
+  },
+  onMatch: change => {
+    return replaceUser(change.doc)
+      .then(() => true)
+      .catch(err => {
+        change.doc.user_for_contact.replaced.status = UserCreationStatus.ERROR;
+        err.changed = true;
+        throw err;
+      });
+  }
```

Comment:
Since this is designed to run in Sentinel only, please add the `asynchronousOnly` property and set it to `true`.

Reply:
Done.  Can you help me understand what this means so that I can include it in the [transitions docs](https://docs.communityhealthtoolkit.org/core/overview/transitions/)?

Reply:
Sure. 
This was added in https://github.com/medic/cht-core/commit/df888af9c658df576a3c72f101fd41a8a488ae0f . Before, transitions were only run in Sentinel, this created additional delays for SMSs, because multiple back-and-forth messages between gateway and API were required to get a response to the submitting user. With the change, transitions are run synchronously over the fresh received SMS, so that no additional requests are required to get responses to users. 

However, there are some transitions that are either very resource intensive (because they need to change many documents, like muting) or are designed (or required) to only work in Sentinel (like this transitions).
So, when loading enabled transitions in API: https://github.com/medic/cht-core/blob/master/api/src/services/config-watcher.js#L44, we only load the ones that should run synchronously.